src/texteff.cpp

```diff
@@ -125,8 +125,8 @@ void DrawTextEffects(DrawPixelInfo *dpi)
 
 	for (TextEffect &te : _text_effects) {
 		if (te.string_id == INVALID_STRING_ID) continue;
-		if (te.mode == TE_RISING || (_settings_client.gui.loading_indicators && !IsTransparencySet(TO_LOADING))) {
-			ViewportAddString(dpi, ZOOM_LVL_OUT_8X, &te, te.string_id, te.string_id - 1, STR_NULL, te.params_1, te.params_2);
-		}
+		if (te.mode == TE_RISING && IsTransparencySet(TO_COSTINCOME)) continue;
+		if (te.mode != TE_RISING && _settings_client.gui.loading_indicators && IsTransparencySet(TO_LOADING)) continue;
```

Comment:
I think the logic got changed.

Before the `loading_indicators` had to be on and it had to be not-transparent to get/keep it shown.
Now you check `loading_indicators` to continue but you inverted the transparent bit.

It might be difficult to see the effect in-game, as with `loading_indicators` disabled no new ones will be made. However, the current ones won't be updated either. So, if you enable `loading_indicators`, put some vehicles in wait-for-load, wait until they are waiting and the indicator is showing. Then pause and disable the loading indicators. When you unpause the indicator remains and won't be updated. I would expect the indicator to disappear immediately instead, like they do in master.

Reply:
```suggestion
		if (te.mode == TE_RISING && IsTransparencySet(TO_COSTINCOME)) continue;
		if (te.mode != TE_RISING && (!_settings_client.gui.loading_indicators || IsTransparencySet(TO_LOADING))) continue;
```

Your absolutely right, silly mistake. This is the correct logic.

(I guess I need to commit this to the branch, then squash to a single tidy commit? Still very much learning git...)